Re-add Chunk Priority/Urgency System

mechoriet opened the pull request:
My attempt to help the Community with porting patches over for 1.17,
this patch got removed cause it quite a big diff

mechoriet committed:
Readded Chunk Urgency System from removed folder

mechoriet committed:
revert removal of dumpChunkInfo depth change

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -117,38 +79,28 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
--    final ChunkPos pos; // Paper - private -> package
 +    volatile int queueLevel; public final int getCurrentPriority() { return queueLevel; } // Paper - OBFHELPER - make volatile since this is concurrently accessed
-+    public final ChunkPos pos; // Paper - private -> public
+     public final ChunkPos pos; // Paper - package->public
```
nothing is changed here, drop

mechoriet replied:
also the volatile or only the ChunkPos?

mechoriet committed:
Resolved requested changed and nuked OBFHelper in favor of getQueueLevel

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -117,38 +79,31 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
 +import it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap; // Paper
```
no longer needed

mechoriet replied:
that import is need for L121 `public final Long2ObjectOpenHashMap<Integer> neighborPriorities = new Long2ObjectOpenHashMap<>();` could inline it

JRoy replied:
yeah that's what I mean when I say that, this can all be inlined now since there is no `it` class

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -117,38 +79,31 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
- 
+     // Paper start - no-tick view distance
+     public final LevelChunk getSendingChunk() {
++
```
drop whitespace change

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -259,91 +214,141 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
          return this.queueLevel;
      }
  
 +    private void setPriority(int i) { setQueueLevel(i); } // Paper - OBFHELPER
```
remove obfhelper

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -259,91 +214,141 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
-         if (playerchunk_state.isOrAfter(ChunkHolder.FullChunkStatus.BORDER) && !playerchunk_state1.isOrAfter(ChunkHolder.FullChunkStatus.BORDER)) {
+-        if (playerchunk_state.isOrAfter(ChunkHolder.FullChunkStatus.BORDER) && !playerchunk_state1.isOrAfter(ChunkHolder.FullChunkStatus.BORDER)) {
 -            this.getFutureIfPresentUnchecked(ChunkStatus.FULL).thenAccept((either) -> {
++        if (playerchunk_state.isOrAfter( FullChunkStatus.BORDER) && !playerchunk_state1.isOrAfter( FullChunkStatus.BORDER)) {
```
remove spaces before `FullChunkStatus`'s :P

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -259,91 +214,141 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
+-        boolean flag2 = playerchunk_state.isOrAfter(ChunkHolder.FullChunkStatus.BORDER);
+-        boolean flag3 = playerchunk_state1.isOrAfter(ChunkHolder.FullChunkStatus.BORDER);
++        boolean flag2 = playerchunk_state.isOrAfter( FullChunkStatus.BORDER);
++        boolean flag3 = playerchunk_state1.isOrAfter( FullChunkStatus.BORDER);
```
spaces

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -259,91 +214,141 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
-                     LevelChunk fullChunk = either.left().get();
-                     ChunkHolder.this.isFullChunkReady = true;
-                     fullChunk.playerChunk = ChunkHolder.this;
-+                    this.chunkMap.distanceManager.clearPriorityTickets(pos);
```
why is this dropped?

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -259,91 +214,141 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
++            this.scheduleFullChunkPromotion(chunkStorage, this.tickingChunkFuture, executor, FullChunkStatus.TICKING);
              // Paper start - cache ticking ready status
--            this.tickingChunkFuture = chunkStorage.postProcess(this); this.tickingChunkFuture.thenAccept((either) -> {
-+            this.tickingChunkFuture = chunkStorage.postProcess(this); ensureMain(this.tickingChunkFuture).thenAccept((either) -> { // Paper - ensure main
```
why is this dropped

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -259,91 +214,141 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
++            this.scheduleFullChunkPromotion(chunkStorage, this.entityTickingChunkFuture, executor, FullChunkStatus.ENTITY_TICKING);
              // Paper start - cache ticking ready status
--            this.entityTickingChunkFuture = chunkStorage.getEntityTickingRangeFuture(this.pos); this.entityTickingChunkFuture.thenAccept((either) -> {
-+            this.entityTickingChunkFuture = chunkStorage.getEntityTickingRangeFuture(this.pos); ensureMain(this.entityTickingChunkFuture).thenAccept((either) -> { // Paper ensureMain
```
same

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -259,91 +214,141 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
+         this.onLevelChange.onLevelChange(this.pos, this::getQueueLevel, this.ticketLevel, this::setQueueLevel);
 +        // Paper start - raise IO/load priority if priority changes, use our preferred priority
-+        priorityBoost = chunkMap.distanceManager.getChunkPriority(pos);
++        priorityBoost = this.chunkMap.getDistanceManager().getChunkPriority( pos );
```
remove whitespace

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -352,83 +357,68 @@ index d907872d80f840b343419f49a6708082da6f921b..ce320672d7602c94dd75ad857435dca6
  import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
  import it.unimi.dsi.fastutil.longs.Long2ByteMap;
  import it.unimi.dsi.fastutil.longs.Long2ByteOpenHashMap;
 +import it.unimi.dsi.fastutil.longs.Long2IntOpenHashMap; // Paper
```
can be removed

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -631,109 +622,90 @@ index 8070acde38c47c364c1d26ec3b7d65da037554a5..7a1f6d1807757a43a7aa471db651404c
+     final ProcessorHandle<ChunkTaskPriorityQueueSorter.Message<Runnable>> ticketThrottlerInput;
+     final ProcessorHandle<ChunkTaskPriorityQueueSorter.Release> ticketThrottlerReleaser;
+-    final LongSet ticketsToRelease = new LongOpenHashSet();
++    final LongSet ticketsToRelease = new LongOpenHashSet(); public final LongSet getOnPlayerTicketAddQueue() { return ticketsToRelease; } // Paper - OBFHELPER
```
can be removed (and made public if needed)

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -857,28 +829,36 @@ index c9b4025f6c3d1be7bca2ff7337dd86e37d21b53e..e41f388e8350010a471410436adf15a9
+         }
  
-     class PlayerTicketTracker extends DistanceManager.FixedPlayerDistanceChunkTracker {
++        protected final int getChunkLevel(long i) { return getLevel(i); } // Paper - OBFHELPER
```
drop obfhelper

JRoy commented on patches/server/0468-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch:
```diff
@@ -857,28 +829,36 @@ index c9b4025f6c3d1be7bca2ff7337dd86e37d21b53e..e41f388e8350010a471410436adf15a9
+     private class PlayerTicketTracker extends DistanceManager.FixedPlayerDistanceChunkTracker {
  
 -        private int viewDistance = 0;
 +        private int viewDistance = 0; private int getViewDistance() { return viewDistance; } private void setViewDistance(int value) { this.viewDistance = value; } // Paper - OBFHELPER
```
drop obfhelper

mechoriet committed:
part one of requested changes

mechoriet committed:
might be safer

mechoriet committed:
compile fix

mechoriet committed:
made the patch way smaller
for some strange my idea found it funny to ChunkHolder.FullChunkStatus.xyz to FullChunkStatus.xyz and blew up the patch for 0 reason,
cause nothing was changed there
